crowbar_framework/app/controllers/barclamp_controller.rb

```diff
@@ -114,12 +114,17 @@ def versions
   def transition
     id = params[:id]       # Provisioner id
     state = params[:state] # State of node transitioning
-    name = params[:name] # Name of node transitioning
+    name = params[:name] # Name or alias of node transitioning
 
-    unless valid_transition_states.include?(state)
-      render text: "State '#{state}' is not valid.", status: 400
-    else
-      status, response = @service_object.transition(id, name, state)
+    node = Node.find_node_by_name_or_alias(name)
+
+    if node.nil?
+      render text: "Node does not exist.", status: 404
+      return
```

Comment:
I think you should put the `return` in front of `render` to get the same behavior as with ruby's "implicit" return values.

Reply:
Even better would be to raise an exception like in other controllers:
`raise ActionController::RoutingError.new("Not Found") if node.nil?`

The exception has common handling and will display proper error message.